exploring.rst

```diff
@@ -0,0 +1,68 @@
+.. _exploring:
+
+Exploring the CPython's Internals
+=================================
+
+This is a quick guide for who want to know about CPython's internals,
+and provide a summary about the source code structure and references.
+
+
+CPython Source Code Typical Layout
+---------------------------------------
+
+For Python modules, the typical layout is:
+
+* ``Lib/<module>.py``
+* ``Modules/_<module>module.c`` (if there's also a C accelerator module)
+* ``Lib/test/test_<module>.py``
+* ``Doc/library/<module>.rst``
+
+For extension-only modules, the typical layout is:
+
+* ``Modules/<module>module.c``
+* ``Lib/test/test_<module>.py``
+* ``Doc/library/<module>.rst``
+
+For builtin types, the typical layout is:
+
+* ``Objects/<builtin>object.c``
+* ``Lib/test/test_<builtin>.py``
+* ``Doc/library/stdtypes.rst``
+
+For builtin functions, the typical layout is:
+
+* ``Python/bltinmodule.c``
+* ``Lib/test/test_<builtin>.py``
+* ``Doc/library/functions.rst``
+
+Some Exception:
+
+* builtin type ``int`` is at ``Objects/longobject.c``
+* builtin type ``str`` is at ``Objects/unicodeobject.c``
+
+CPython Internals Tutorials
+---------------------------
+
+There are several materials about CPython internals on the Internet:
+
+* Python 2
```

Comment:
@brettcannon @willingc I would like to keep these article in the list, each of them provides some different point of view of CPython. Guido's article is a fast glimpse for REPL, but no other such as PythonObjects, parser etc., the oldest version of CPython comes from Philip Guo `CPython internals`, is using `2.7.8`, but it is still a good entry video (possibly the only) for newcomer to know about the big picture of CPython.